carwings.net/Carwings.cs

```diff
@@ -32,80 +33,140 @@ public Carwings()
             };
         }
 
-        public async Task<Vehicle[]> Login(string userid, string password)
+        /// <summary>
+        /// Log in to the specified Nissan account
+        /// </summary>
+        /// <param name="userid">Email address</param>
+        /// <param name="password">password</param>
+        /// <param name="country">Country or Nissan region name, different from Nissan's V1 API.  Valid values are USA: "US"</param>
+        /// <param name="cancellationToken">Support for cancellation.</param>
+        /// <returns>An array of Vehicles for this user.  If the login credentials were wrong, it returns a null Vehicle[].</returns>
+        [Obsolete("Provide a cancellation token")]
+        public async Task<Vehicle[]> Login(string userid, string password, string country)
+        {
+            return await Login(userid, password, country, CancellationToken.None);
+        }
+
+        /// <summary>
+        /// Log in to the specified Nissan account
+        /// </summary>
+        /// <param name="userid">Email address</param>
+        /// <param name="password">password</param>
+        /// <param name="country">Country or Nissan region name, different from Nissan's V1 API.  Valid values are USA: "US"</param>
+        /// <param name="cancellationToken">Support for cancellation.</param>
+        /// <returns>An array of Vehicles for this user.  If the login credentials were wrong, it returns a null Vehicle[].</returns>
+        public async Task<Vehicle[]> Login(string userid, string password, string country, CancellationToken cancellationToken)
```

Comment:
usually the convention is to support a default value for the cancellation token so that not everyone has to pass

Reply:
Hmmm.  Yeah, ok...  I should leave in the overloads then, minus the obsolete attribute.